rsc.js

```diff
@@ -1,47 +1,27 @@
 // @ts-check
 
-// cjs-module-lexer will let all of react's (named) exports through unchanged.
-module.exports = { ...require("react") };
+// missing functions
+module.exports.createContext = polyfillMissingFn("createContext");
+module.exports.createFactory = polyfillMissingFn("createFactory");
+module.exports.act = polyfillMissingFn("act");
+module.exports.unstable_act = polyfillMissingFn("unstable_act");
+module.exports.unstable_useCacheRefresh = polyfillMissingFn("unstable_useCacheRefresh");
+module.exports.useContext = polyfillMissingFn("useContext");
+module.exports.useDeferredValue = polyfillMissingFn("useDeferredValue");
+module.exports.useEffect = polyfillMissingFn("useEffect");
+module.exports.useImperativeHandle = polyfillMissingFn("useImperativeHandle");
+module.exports.useInsertionEffect = polyfillMissingFn("useInsertionEffect");
+module.exports.useLayoutEffect = polyfillMissingFn("useLayoutEffect");
+module.exports.useReducer = polyfillMissingFn("useReducer");
+module.exports.useRef = polyfillMissingFn("useRef");
+module.exports.useState = polyfillMissingFn("useState");
+module.exports.useSyncExternalStore = polyfillMissingFn("useSyncExternalStore");
+module.exports.useTransition = polyfillMissingFn("useTransition");
+module.exports.useOptimistic = polyfillMissingFn("useOptimistic");
```

Comment:
I think instead of having an array of exports followed by a loop and then another piece of code with never-existing, explicit, exports we can also just explicitly move them to the top and only mention them once that way.

Reply:
i like mentioning them once! there's one behavioral change here tho: in my PR i used `||=` to assign the stubs, so that if  something was present, it wouldn't get replaced. which is important bc weight get regular react, not the `react-server` one. so ig the question is: will `cjs-module-lexer` understand this?

```
module.export.useState ||= polyfillMissingFn('useState')
```

Reply:
I believe it's not necessary because we changed the order here:

We set these exports first, and then further down we `Object.assign(module.exports, require("react"))`, which will overwrite all properties of `module.exports` with the values from `require("react")`.

Reply:
yeah i see now. yeah i think this makes sense!